cli/util/env_manager.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::env;
+use std::path::Path;
+use std::sync::Arc;
+use std::sync::Mutex;
+use std::sync::OnceLock;
+
+use deno_terminal::colors;
+
+#[derive(Debug, Clone)]
+struct EnvManagerInner {
+  // Track all loaded variables and their values
+  loaded_variables: HashSet<String>,
+  // Track variables that are no longer present in any loaded file
+  unused_variables: HashSet<String>,
+  // Track original env vars that existed before we started
+  original_env: HashMap<String, String>,
+}
+
+impl EnvManagerInner {
+  fn new() -> Self {
+    // Capture the original environment state
+    let original_env: HashMap<String, String> = env::vars().collect();
+
+    Self {
+      loaded_variables: HashSet::new(),
+      unused_variables: HashSet::new(),
+      original_env,
+    }
+  }
+}
+
+#[derive(Debug, Clone)]
+pub struct EnvManager {
+  inner: Arc<Mutex<EnvManagerInner>>,
+}
+
+// Global singleton instance
+static ENV_MANAGER: OnceLock<EnvManager> = OnceLock::new();
+
+impl EnvManager {
+  /// Get the global singleton instance
+  pub fn instance() -> &'static EnvManager {
+    ENV_MANAGER.get_or_init(|| EnvManager {
+      inner: Arc::new(Mutex::new(EnvManagerInner::new())),
+    })
+  }
+
+  // Internal method that accepts an already-acquired lock to avoid deadlocks
+  fn load_env_file_inner<P: AsRef<Path>>(
+    &self,
+    file_path: P,
+    log_level: Option<log::Level>,
+    inner: &mut EnvManagerInner,
+  ) {
+    let path_str = file_path.as_ref().to_string_lossy().to_string();
```

Comment:
Recommend to change the function to receive a resolved `&Path`, then use `file_path.display()` in the eprintlns.